src/server/pkg/obj/testing/util.go

```diff
@@ -0,0 +1,82 @@
+package testing
+
+import (
+	"os"
+	"testing"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/require"
+	"github.com/pachyderm/pachyderm/src/server/pkg/obj"
+)
+
+func NewDefaultAmazonConfig() *obj.AmazonAdvancedConfiguration {
+	return &obj.AmazonAdvancedConfiguration{
+		Retries:        obj.DefaultRetries,
+		Timeout:        obj.DefaultTimeout,
+		UploadACL:      obj.DefaultUploadACL,
+		Reverse:        obj.DefaultReverse,
+		PartSize:       obj.DefaultPartSize,
+		MaxUploadParts: obj.DefaultMaxUploadParts,
+		DisableSSL:     obj.DefaultDisableSSL,
+		NoVerifySSL:    obj.DefaultNoVerifySSL,
+		LogOptions:     obj.DefaultAwsLogOptions,
+	}
+}
+
+func LoadAmazonParameters(t *testing.T) (string, string, string, string) {
+	id := os.Getenv("AMAZON_DEPLOYMENT_ID")
+	secret := os.Getenv("AMAZON_DEPLOYMENT_SECRET")
+	bucket := os.Getenv("AMAZON_DEPLOYMENT_BUCKET")
```

Comment:
A few high-level questions:
- Do we create a new AWS bucket for each test run? Is that in this PR (and I just missed it) or is that separate?
- IIUC, this is all passed in via env vars so that we can run it in CI without exposing our cloud provider creds? Is that right? By extension, we'll have to set up these env vars in e.g. Travis?

Reply:
* creating a new bucket is hard and it makes cleanup harder, so these just use preconfigued buckets
  * this lets us pare down our creds to the minimal set of permissions to R/W access in one bucket, which is pretty nice
  * I'm not certain that there won't be flakiness from multiple deployments using the same bucket, but if it comes up we can add namespacing to our object clients.
* Yep - I'm trying to get this running in CircleCI at the moment - the creds are also in 1password